code/obj/item/cigarette.dm

```diff
@@ -1122,10 +1122,11 @@
 		return
 
 	afterattack(atom/target, mob/user as mob)
-		if (!on && istype(target, /obj/reagent_dispensers/fueltank))
+		if (!on && (istype(target, /obj/reagent_dispensers/fueltank) || istype(target, /obj/item/reagent_containers/food/drinks/fueltank)))
 			if (src.fuel == -1)
 				user.show_text("You can't seem to find any way to add more fuel to [src]. It's probably fine.", "blue")
 				return
+			//Even if it is a handheld fuel tank, this will behave the same.
 			var/obj/reagent_dispensers/fueltank/O = target
 			var/fuelamt = O.reagents.get_reagent_amount("fuel")
```

Comment:
Even though it works (by guise of most/all objects having a reagents var) it is sort of a compiler trick and for code that runs very sporadically I wonder whether just having two separate condition paths might be better as a simple solution, or if something even better might come to mind (though not for me).
Maybe some more people might chime in with what they think.

Reply:
Do you mean one if statement that treats the regular fuel tank as a reagent holder, and one that treats the handheld one as having volume?

Also, I had a couple other questions. does anyone know why the zippo fills up by 50 instead of O.amount_per_transfer_from_this, and why the zippo doesn't have a cap to how much fuel it can hold?

Reply:
What i'm refering to specifically is casting the food/drinks/fueltank into a reagent_dispensers/fueltank.
They're two completely different objects/types, the code happens to work due to polymorphism and them both containing a reagents datum, but it feels kind of wrong as it's treating an item as a completely different and barely related (outside of a common ancestor in obj/) one.

Regarding the filling up by 50 constantly, I assume it was laziness when someone first added the functionality, the fire extinguisher suffers from the same and they both probably should be addressed as a separate change sometime. Same for the lack of a cap on the lighter.

Reply:
Okay, I'll work on that tomorrow and fix those other 2 issues while I'm at it.

Reply:
Wait, actually I'm confused. Isn't `var/obj/reagent_dispensers/fueltank/O = target` already casting the food/drinks/fueltank into a reagent_dispensers/fueltank? The refill happens when the object is either of those, but target is always treated as reagent_dispensers/fueltank. Am I missing something here?

Reply:
Original functionality checks for whether the target is of a reagent_dispensers/fueltank type, to see if it's a fueltank object and when it is, casts that object into a variable of that type (reagent_dispensers/fueltank) to be able to access its reagents datum and do further operations from there.

Your code extends the condition check to also support the handheld food/drinks/fueltank small fueltank. However, the remainder of the code still casts that handheld fuel tank into a variable of the type of the large reagent dispenser fueltank. It's casting the object into something that it isn't in reality. It works in this case because they both inherit from /obj/ and so they both have a reagents datum which is being accessed, but the method of doing so is really weird and hacky.
In shorter summation: you're casting a small fueltank into a completely different object than it really is. It manages to work due to some similarities in the obj ancestor, but it is a very weird approach and I think could be prone to breaking if something changed.

Reply:
Yeah what Hazo said. If you have questions, feel free to ping me in #imcoder!